### ISSUE TITLE
We have updated some dependencies in pyworkflow.... and conflicts with some here.

### ISSUE THREAD

**Issue by pconesa** [DavidHerreros/xmipp_metadata#20]
Hi David, recently we have updated pyworkflow dependencies.....and some conflicts with some here:

xmipp-metadata 1.0.13 requires numpy<=1.23.5, but you have numpy 1.24.4 which is incompatible.
xmipp-metadata 1.0.13 requires pillow<=9.5.0; python_version >= "3.7", but you have pillow 10.1.0 which is incompatible.

Ideally, no plugin should define versions for dependencies already specified in pyworkflow or scipion-em.

But I understand that this package is also installed in a separate environment and in that case those dependencies are needed.

A possible solution would be to limit the lower version but not the higher, allowing pyworkflow/scipion-em to increase the version freely.

Another solution would be to release a new version updating the  higher version of those in conflict but this will trigger again in the next dependencies update. 

Any thoughts?

**Comment by DavidHerreros** [comment]
Hi @pconesa,

In principle, this package is updated and should be able to install those requirements in the last uploaded version (which should be `1.1.3`). Here are the new requirements for this plugin:

```
numpy<=1.26.4
pillow==8.4.0; python_version == '3.6'
pillow<=10.1.0; python_version >= '3.8'
pandas<=2.1.4
starfile!=0.5.0
mrcfile
emfile
emtable
scikit-image<=0.24.0
morphsnakes
joblib
pytest
```

The problem with this package is that it cannot install a higher version of pandas that `2.1.4`, as it introduces some changes that make the current operations to handle the metadata incompatible. This should be reviewed in the future, although I would not like to do it unless it is completly necessary.

Let me know if updating the package works, we can modify the requirements to make it more flexible.

Also, thanks for the report!! 😉

**Comment by pconesa** [comment]
Thanks David! Does this package installs in Scipion environment?

**Comment by DavidHerreros** [comment]
> Thanks David! Does this package installs in Scipion environment?.

Hi Pablo!

Do you mean if I have it installed in my Scipion environment? If that is the case, I have it installed... but my environment is not up to dat with the new dependencies merged in master/devel...

If you mean if there are any plugins defining this package as a dependency, I think the only one should be Flexutils

**Comment by pconesa** [comment]
Yes I mean if it is a dependency of the plugin and therefore installed inside Scipion env or if it was installed in a dedicated environment?
________________________________
De: DavidHerreros ***@***.***>
Enviado: lunes, 12 de agosto de 2024 17:45
Para: DavidHerreros/xmipp_metadata ***@***.***>
Cc: Pablo Conesa ***@***.***>; Mention ***@***.***>
Asunto: Re: [DavidHerreros/xmipp_metadata] We have updated some dependencies in pyworkflow.... and conflicts with some here. (Issue #20)


Thanks David! Does this package installs in Scipion environment?.

Hi Pablo!

Do you mean if I have it installed in my Scipion environment? If that is the case, I have it installed... but my environment is not up to dat with the new dependencies merged in master/devel...

If you mean if there are any plugins defining this package as a dependency, I think the only one should be Flexutils

**Comment by DavidHerreros** [comment]
In that case yes, it is installed in the environment by Flexutils plugin to avoid using Xmipp to read/write metadata and binary files
